src/NumericalAlgorithms/Spectral/SwshTransform.hpp

```diff
@@ -34,19 +35,275 @@ namespace detail {
+// `SwshDerivatives.hpp`
+template <typename Transform, typename FullTagList>
+struct dispatch_to_transform;
 }  // namespace detail
 
 /*!
  * \ingroup SwshGroup
- * \brief A class which gathers all necessary shared structure among several
- * spin-weighted spherical harmonic transforms and dispatches to libsharp. Each
- * `TransformJob` represents exactly one spin-weighted transform libsharp
- * execution call, and one inverse transform libsharp execution call.
+ * \brief Perform a forward libsharp spin-weighted spherical harmonic transform
+ * on any number of supplied `SpinWeighted<ComplexDataVector, Spin>`.
+ *
+ * \details This function places the result in one or more
+ * `SpinWeighted<ComplexModalVector, Spin>` returned via provided
+ * pointer.  This is a simpler interface to the same functionality as the
+ * \ref DataBoxGroup mutation compatible `SwshTransform`.
  *
+ * The following parameter ordering for the multiple input interface is enforced
+ * by the interior function calls, but is not obvious from the explicit
+ * parameter packs in this function signature:
+ *
+ * - `size_t l_max` : angular resolution for the transform
+ * - `size_t number_of_radial_points` : radial resolution (number of consecutive
+ * transforms in each of the vectors)
+ * - any number of `gsl::not_null<SpinWeighted<ComplexModalVector,
+ * Spin>*>`, all sharing the same `Spin` : The return-by-pointer containers for
+ * the transformed modal quantities
+ * - the same number of `const SpinWeighted<ComplexDataVector, Spin>&`, with the
+ * same `Spin` as the previous function argument set : The input containers of
+ * nodal spin-weighted spherical harmonic data.
+ *
+ * template parameters:
+ * - `Representation`: Either `ComplexRepresentation::Interleaved` or
+ * `ComplexRepresentation::RealsThenImags`, indicating the representation for
+ * intermediate steps of the transformation. The two representations will give
+ * identical results but may help or hurt performance depending on the task.
+ * If unspecified, defaults to `ComplexRepresentation::Interleaved`.
+ *
+ * The result is a set of libsharp-compatible coefficients.
+ * \see SwshTransform for more details on the mathematics of the libsharp
+ * data structures.
+ *
+ * \warning The collocation data is taken by const reference, but can be
+ * temporarily altered in-place during intermediate parts of the computation.
+ * The input data is guaranteed to return to its original state by the end of
+ * the function. In a setting in which multiple threads access the same data
+ * passed as input to this function, a lock must be used to prevent access
+ * during the execution of the transform.
+ */
+template <
+    ComplexRepresentation Representation = ComplexRepresentation::Interleaved,
+    int Spin, typename... ModalThenNodalTypes>
+void swsh_transform(
+    const size_t l_max, const size_t number_of_radial_points,
+    gsl::not_null<SpinWeighted<ComplexModalVector, Spin>*> first_coefficient,
```

Comment:
I sure wish I could do something like that. It would make the interface so much cleaner, but the c++ template function inference appears to be too 'greedy' to have multiple parameter packs of different types in sequence. In other places this is avoided by, for instance, a parameter pack of types in a struct template to guide the member function inference.

Happy to add the `const` tho :P